File: Aula_09/Projeto_1/ui.py
from cliente import Cliente, Clientes

class UI:
    def menu():
        print("1 - Inserir cliente, 2 - Listar clientes, 3 - Atualizar cliente, 9 - Fim")
        return int(input("Informe uma opção: "))

    def main():
        op = 0
        while op != 9:
            op = UI.menu()
            if op == 1:
                UI.inserir_cliente()
            if op == 2:
                UI.listar_clientes()
            if op == 3:
                UI.atualizar_clientes()

    @classmethod
    def inserir_cliente(cls):
        # id = int(input("Informe o id: "))
        nome = input("Infome o nome: ")
        email = input("Informe o email: ")
        fone = input("Informe o fone: ")
        cliente = Cliente(0, nome, email, fone)
        Clientes.inserir(cliente)

    @classmethod
    def listar_clientes(cls):
        clientes = Clientes.listar()
        if len(clientes) == 0:
            print("Nenhum cliente cadastrado")
        else:
            for cliente in clientes:
                print(cliente)

    @classmethod
    def atualizar_clientes(cls):
        cls.listar_clientes()
        id = int(input("Informe o id do cliente a ser alterado: "))
        nome = input("Informe o novo nome: ")
        email = input("Informe o novo email: ")
        fone = input("Informe o novo fone: ")
        cliente = Cliente(id, nome, email, fone)
        Clientes.atualizar(cliente)

UI.main()

File: Aula_09/Projeto_1_java/Ui.java
import java.util.List;
import java.util.Scanner;

public class Ui{
    static Scanner scanner = new Scanner(System.in);

    public static int menu(){
        System.out.println("Cadastro de Clientes");
        System.out.println("1 - Inserir, 2 - Listar, 3 - Atualizar, 4 - Excluir, 9 - Fim");
        System.out.print("Informe uma opção: ");
        int op = scanner.nextInt();
        scanner.nextLine();
        return op;
    }

    public static void main(String[] args){
        int op = 0;
        while (op != 9){
            op = menu();
            switch (op){
                case 1:
                    inserirCliente();
                    break;
                case 2:
                    listarClientes();
                    break;
                case 3:
                    atualizarCliente();
                    break;
                case 4:
                    excluirCliente();
                    break;
                case 9:
                    System.out.println("Saindo...");
                    break;
                default:
                    System.out.println("Opção ivalida. Tente novamente.");
            }
        }
    }

    public static void inserirCliente(){
        System.out.print("Informe o nome: ");
        String nome = scanner.nextLine();

        System.out.print("Informe o email: ");
        String email = scanner.nextLine();

        System.out.print("Informe o fone: ");
        String fone = scanner.nextLine();

        Cliente cliente = new Cliente(0, nome, email, fone);

        Clientes.inserir(cliente);
    }

    public static void listarClientes(){
        List<Cliente> clientes = Clientes.listar();

        if (clientes.isEmpty()){
            System.out.println("Nenhum cliente cadastrado.");
        } else {
            for (Cliente cliente : clientes){
                System.out.println(cliente);
            }
        }
    }

    public static void atualizarCliente(){
        listarClientes();

        System.out.print("Informe o id do cliente a ser alterado: ");
        int id = scanner.nextInt();
        scanner.nextLine();

        System.out.print("Informe o novo nome: ");
        String nome = scanner.nextLine();

        System.out.print("Informe o novo email: ");
        String email = scanner.nextLine();

        System.out.print("Informe o novo fone: ");
        String fone = scanner.nextLine();

        Cliente cliente = new Cliente(id, nome, email, fone);

        Clientes.atualizar(cliente);
    }

    public static void excluirCliente(){
        listarClientes();

        System.out.print("Informe o id do cliente a ser excluido: ");
        int id = scanner.nextInt();

        Cliente cliente = new Cliente(id, "", "", "");

        Clientes.excluir(cliente);
    }
}